### ISSUE TITLE
Berkeley Mapper for casual users?

### ISSUE THREAD

**Issue by cjconroy** [ArctosDB/arctos#5587]
An instructor at UC Berkeley wrote to ask if Berkeley Mapper will be available to casual users. He has a biology lab where he has students search up various specimens and then map them. However, it looks like Berkeley mapper is not available unless you create an account. This class might be hundreds of students. What do you recommend?  Creating hundreds of accounts that will likely be used for only a week, or perhaps the instructor creates one account for him and gives all the students access to it. Or could BerkeleyMapper be given back to casual users like before?
Thanks,
Chris
p.s. he also wondered why Arctos is often very slow, or unresponsive. I said that's just how it is.

**Comment by dustymc** [comment]
I don't think I have any strong preferences or need to keep things internal, especially now that we have some light authentication, and that's easy to change.

Any reason not to make BM public? (And how DO we make such decisions?)

Should anything else be changed?

Here's the widget code, which includes permissions - let me know if anything needs 'translated.'


```

<select form="cat_rec_sch_frm" name="tools_ctl" id="tools_ctl" size="1">
		<option value="">[ Tools ]</option>
		<cfif listfindnocase(session.roles,'manage_transactions') and len(add_to_trans_id) gt 0>
			<option value="/info/loan_item_pick.cfm?add_to_trans_id=#add_to_trans_id#">Add items to Transaction</option>
		</cfif>
		<cfif len(session.username) gt 0>
			<option value="/bnhmMaps/bnhmMapData.cfm">BerkeleyMapper</option>
			<option value="/SpecimenResultsDownload.cfm">Download</option>
			<option value="reloadAtURL">Reload with shareable URL</option>
			<option value="/info/cat_record_group.cfm">Summarize</option>
			<option value="annotateall">Comment or report Bad Data</option>
			<option value="savsearch">Save Search</option>
			<option value="archiveRecords">Archive Search</option>
		<cfelse>
			<option value="login">Log in or create account</option>
		</cfif>
		<cfif listfindnocase(session.roles,'coldfusion_user')>
			<option value="/Reports/report_printer.cfm">Arctos Reporter</option>
			<cfif listfindnocase(session.roles,'manage_records')>
				<option value="/multiAttribute.cfm">Manage Attributes</option>
				<option value="/place.cfm??sch=collecting_event">Manage Collecting Event</option>
				<option value="/Encumbrances.cfm">Manage Encumbrances</option>
				<option value="/multiIdentification.cfm">Manage Identifications</option>
				<option value="/multiAgent.cfm">Manage Collectors</option>
				<option value="/tools/replace_agents.cfm">Manage Agents</option>
				<option value="/findContainer.cfm">Part Locations</option>
				<option value="/place.cfm?sch=locality">Manage Locality</option>
				<option value="/SpecimenUsage.cfm?action=search">Manage Publications</option>
				<option value="/bulkSpecimenEvent.cfm">Manage Specimen Events</option>
				<option value="/addAccn.cfm">Manage Accession</option>
				<option value="/tools/bulkPart.cfm">Modify Parts</option>
				<option value="/tools/magicEntity.cfm">Magic Entities</option>
				<option value="/Reports/specrescollevent.cfm">Download Collecting Event CSV</option>
				<option value="/info/part_data_download.cfm">Download Parts</option>
				<option value="/tools/identifier_download.cfm">Download Identifiers</option>
				<option value="/Reports/attribute_data_download.cfm">Download Attributes</option>
				<option value="/SpecimenResultsDownload.cfm?action=bulkloaderFormat">Download for Record Bulkloader</option>
				<option value="/SpecimenResultsDownload.cfm?action=citationFormat">Download for Citation Bulkloader</option>
			</cfif>
			<option value="/info/checkLegalTaxonomy.cfm">Check Arctos Legal (taxonomy)</option>
			<option value="/SpecimenUsage.cfm?tbl_use=sensu&action=search">View Sensu Publications</option>
		</cfif>
	</select>

```

I believe 'slow' is related to https://github.com/ArctosDB/internal/issues/187 - I don't see 'slow' but there's plenty of 'lost in transit' which probably isn't distinguishable to users. TACC is aware, I have no idea what else to do about that.

**Comment by campmlc** [comment]
I strongly support making BM public - it is one of
Arctos's most important tools for the general public.
In this case, I also support allowing the instructor to use a single login
for the several hundred students for his class - perhaps suggest the login
is something with the course name and date. Yes, it would be great if we
could get info on the number of students etc etc, but the realities of
designing a course assignment may lead to instructors dropping any use of
Arctos in an educational setting if we make it too onerous. Or, we can
require the login and ask this guy for his feedback - maybe it does work.
But I'd hate to go in to teach a class not having a good idea if the
assignment will work in the time allotted or not.

**Comment by dustymc** [comment]
>allowing the instructor to use a single login for the several hundred students

That is (still) NOT compatible with the structure or design of Arctos, and will result in an unpleasant experience for everyone. (Unless you're supporting it by writing a UI for them to use?!)

**Comment by Jegelewicz** [comment]
OK you guys - making "BM" public - is that really a good idea? ;-)

Berkeley Mapper seems like something we would want everyone to be able to access. Unless it involves some significant use of resources, I think it should just be there for everyone (logged in or not).

No to the single username and password for hundreds of people - that is asking for trouble eventually. Someone could do it, but I would not be recommending it. When those hundred students are all hitting up Arctos from the classroom as a single user, don't they run the risk of getting blocked? (Maybe not if they are all coming from different IP addresses, but shouldn't that also be suspicious?)

**Comment by campmlc** [comment]
I am OK with all this - and I do prefer the idea of Berkeley Mapper being
public, rather than BM :)
BUT - maybe we need a tab or something for Educators on our portal - where
we make some suggestions, such as - your students will need a login before
accessing Arctos, perhaps have them do this prior to coming to class, and
no, we do not allow class-level logins. For questions, please contact etc
etc.
Right now we have resources for people who want to search, and for
curatorial tools. Why not add an Educators tool? With links to BLUE, AIMUP
modules etc?

**Comment by dustymc** [comment]
> involves some significant use of resources

It does but I think we're OK. (And hopefully that's going to be getting more efficient anyway.)

>asking for trouble eventually.

They'd all be doing conflicting things, it would not be a good experience even if they didn't get blocked by Arctos and/or TACC.

>your students will need a login before accessing Arctos

Are you suggesting policy or ????

**Comment by campmlc** [comment]
I'm just suggesting a specific tab or tool titled "For Educators" with a link to a documentation page and some resources and suggestions. There are obviously people out there using Arctos now who aren't in the loop with all our changes. It would show we acknowledge the needs of that user group and are trying to provide some support.

**Comment by acdoll** [comment]
I support making Berkeley Mapper public however possible - its a great tool and a useful service for us to provide access to. 

We should not, as a general practice, recommend the practice of having multiple users accessing the same account. At a minimum, it will muddy the waters of our usage analytics. If people choose to do so, that is up to them, but we should avoid it. Is there really any harm in having students create their own accounts? Its an easy process and a rather minimal amount of data for us to store and we don't have to keep it forever. 
What is our current procedure for culling users roles? Made inactive after X months, Y years after that their account data is deleted? These actions can be preempted by notification emails - no response = approval to proceed. 

> "For Educators"

Developing some specific interface(s) for educators is a great idea but probably a larger issue for further down the road after we've settled some of our more pressing UI and general functionality priorities. But, yes, even just some links to helpful documentation would be a great start.

**Comment by dustymc** [comment]
>multiple users accessing the same account.

I think this is some leftover from a dead branch of the require accounts thread??? It's not something I'm intending to actively try to block (and I can't speak for TACC on this), but it still sounds like a great recipe for convincing someone that Arctos is broken and/or haunted (as some co-user changes things that affect them) and we should at least have some sort of policy to point to.

>current procedure for culling users roles?

By request of the user, if we can verify with some certainty that they're who they say they are. (I'm up for whatever, but will remain paranoid until/unless there's some policy.)

>specific interface(s) for educators

There's an API....

**Comment by mkoo** [comment]
This seems like a no-brainer to allow BerkeleyMapper views of record results for public users. Same for graphical summary of results. It's just alternative views of search results and what users expect. Arctos is and should be a spatial dtabase and not having spatial views seems to go against basic Arctos. 

Also, users getting this far have clicked on terms of agreement, and BM is an on-call service so not used for every search result. I really dont see the issue.

I have more concern about scoping out the framework a 'For Educators' tools, or requiring logins for every student, or one login account for a class.... if serious suggestion, file another issue

**Comment by dustymc** [comment]
> dont see the issue.

My question is whether we want to make anything else available to no-login users, but that can be addressed in another Issue if necessary - patched.
